File: Lab 2/Alex/createTopic.py
import kafka
from kafka.admin import KafkaAdminClient, NewTopic

# Define client
kafkaClient = KafkaAdminClient(
    bootstrap_servers="0.0.0.0:9093",
)

# Create topic
topics = []
topics.append(NewTopic(name="lab2", num_partitions=3, replication_factor=1))

try:
    #print()
    kafkaClient.create_topics(new_topics=topics, validate_only=False)
except kafka.errors.TopicAlreadyExistsError:
    print("Topic already created")

# Delete Topic(s)
# kafkaClient.delete_topics(["lab2"])

# List Topics
print("Topics:", kafkaClient.list_topics())

File: Lab 2/Alex/createTopic.js
//const Kafka = require("kafkajs").Kafka
const {Kafka} = require("kafkajs")
const topic = "lab2"

run();
async function run(){
    try
    {
        const kafka = new Kafka({
            "brokers" :["localhost:9093"]
        })

        const admin = kafka.admin();
        console.log("Connecting.....")
        await admin.connect()
        console.log("Connected!")
        await admin.createTopics({
            "topics": [{
                "topic" : topic,
                "numPartitions": 3
            }]
        })
        console.log("Created Successfully!")
        await admin.disconnect();
    }
    catch(ex)
    {
        console.error(`Something bad happened ${ex}`)
    }
    finally{
        process.exit(0);
    }


}